lib/src/Pos/Arbitrary/Block.hs

```diff
@@ -215,23 +217,26 @@ instance ( Arbitrary $ SscPayload ssc
          , Arbitrary (SscPayloadDependsOnSlot ssc)
          , SscHelpersClass ssc
          , HasConfiguration
+         , ssc ~ SscGodTossing
          ) =>
-         Arbitrary (T.MsgBlock ssc) where
+         Arbitrary T.MsgBlock where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance T.BiSsc ssc => Buildable (T.BlockHeader ssc, PublicKey) where
+instance T.BiSsc SscGodTossing => Buildable (T.BlockHeader, PublicKey) where
     build (block, key) =
         bprint
             ( build%"\n"%
               build%"\n"
             ) block key
 
-newtype BlockHeaderList ssc = BHL
-    { getHeaderList :: ([T.BlockHeader ssc], [PublicKey])
-    } deriving (Eq)
+newtype BlockHeaderList = BHL
+    { getHeaderList :: ([T.BlockHeader], [PublicKey])
+    }
+
+deriving instance Eq T.BlockHeader => Eq BlockHeaderList
```

Comment:
Why did you change it to standalone deriving?

Reply:
Because previously it was automatically deriving a `Eq (T.BlockHeader ssc)` constraint for the instance, but after I removed `ssc` it broke.

Reply:
It seems very strange that this module doesn't see `Eq T.BlockHeader` instance and this constraint is needed at all.